### PR TITLE
0.8 Docs Changes

### PR DESCRIPTION
**Pull request by stellarsquall**
fixes bad merge in [new-metrics-docs PR 1977](https://github.com/hashicorp/boundary/pull/1977)

## Boundary 0.8 Release: Week of May 6th

🎟️  [Asana](https://app.asana.com/0/563192436488770/1202051708682006/f)
:mag: [Deploy Preview](https://boundary-git-metrics-docs-hashicorp.vercel.app/docs/operations)

This PR: 

- Removes the note under [`audit_enabled`](https://www.boundaryproject.io/docs/configuration/events/overview)
- Adds a new **Metrics** doc based on the [draft](https://docs.google.com/document/d/11cQycnZYk0B2c2N52ae1CfRE06Chfwvuo6wvSqIBwck/edit?usp=sharing)
- Adds a **Health Endpoint** doc
- Updates **Events -> Common Sink Parameters** to describe additional `audit_filter` parameters

boundary-metrics

### What's left to do:
- [x] Add the tutorial title & link under the [**Tutorial** section](https://boundary-git-new-metrics-doc-hashicorp.vercel.app/docs/operations/metrics#tutorial) (@stellarsquall)
- [x] Add a note about [`audit_enabled` parameter](https://boundary-git-new-metrics-doc-hashicorp.vercel.app/docs/configuration/events/overview#audit_enabled) of `events` stanza _if any_ (@sswati0)
- [x] Add release notes for 0.8 (@xingluw)
- [x] Change release section navigation (@xingluw)

### CODE REVIEW

File: website/content/docs/operations/metrics.mdx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+workers.
+
+| Name                                                          | Description                                   |
+|---------------------------------------------------------------|-----------------------------------------------|
+| `boundary_worker_proxy_http_write_header_duration_seconds`    | Histogram of time elapsed after the TLS connection is established to when the first http header is written back from the server. |
+| `boundary_worker_proxy_websocket_active_connections`          | A gauge of the current count of open proxy connections on the worker. |
+| `boundary_worker_proxy_websocket_received_bytes_total`        | Count of received bytes sent over all proxy connections handled by the worker. |
+| `boundary_worker_proxy_websocket_sent_bytes_total`            | Count of sent bytes  sent over all proxy connections handled by the worker. |
+
+
```
Comment (A-440Hz) [inline website/content/docs/operations/metrics.mdx:49]:
```
### Other

The following current miscellaneous metrics provided for Boundary.

| Name                                                          | Description                                   |
|---------------------------------------------------------------|-----------------------------------------------|
| `boundary_build_info`    | A gauge set to 1 with label values detailing the current Go version, git revision, and Boundary version. |
```

Do we want something like this detailing the current build info metric? 

Maybe we take out "with label values" since labels are introduced right after.

Does that mean we will need a label table with the 3 build info labels in the following section? I think that would be extraneous since this description would already get the point across.

@talanknight what do you think?

Comment (tmessi) [reply website/content/docs/operations/metrics.mdx:48]: I think we should be explicit and include both the "Other" block shown above as well as a label table for this metric. And I think we should then include examples of how this can be used along with the other metrics in the TBD tutorial.

Comment (A-440Hz) [reply website/content/docs/operations/metrics.mdx:48]: Sounds good. I wrote a template for the labels block in the [doc draft](https://docs.google.com/document/d/11cQycnZYk0B2c2N52ae1CfRE06Chfwvuo6wvSqIBwck/edit).
For the examples, I think Robin is covering that in his learn guide tutorial

Comment (stellarsquall) [reply website/content/docs/operations/metrics.mdx:55]: @A-440Hz, how does the new Other section look for `boundary_build_info`?

Comment (A-440Hz) [reply website/content/docs/operations/metrics.mdx:55]: Looks good, thanks. I think I made a typo though. Could you correct it to "The following lists the current miscellaneous metrics provided for Boundary"?

File: website/content/docs/operations/metrics.mdx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+layout: docs
+page_title: Boundary Metrics
+description: |-
+  Obtain visibility of various components of a running Boundary
+---
+
+## Boundary Metrics
+
+Boundary provides metrics through the `/metrics` path using a listener with the
+`"ops"` purpose. By default a listener with that purpose runs on port `9203`.
```
Comment (taoism4504) [inline website/content/docs/operations/metrics.mdx:11]:
```suggestion
`"ops"` purpose. By default, a listener with that purpose runs on port `9203`.
```

File: website/content/docs/operations/health.mdx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+layout: docs
+page_title: Boundary Health Endpoints
+description: |-
+  Verify the Boundary controller server is up and able to receive requests
+---
+
+## Boundary Health Endpoints
+
+Boundary provides health monitoring through the `/health` path using a listener
+with the `"ops"` purpose. By default a listener with that purpose runs on port
```
Comment (taoism4504) [inline website/content/docs/operations/health.mdx:11]:
```suggestion
with the `"ops"` purpose. By default, a listener with that purpose runs on port
```

File: website/content/docs/operations/health.mdx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+`9203`. See the [example configuration](#example-configuration) section for an
+example listener stanza in a `config.hcl` file.
+
+### Requirements
+
+For the controller health endpoint to be enabled, any Boundary instance must be
```
Comment (taoism4504) [inline website/content/docs/operations/health.mdx:17]:
```suggestion
To enable the controller health endpoint, any Boundary instance must be
```

File: website/content/docs/operations/health.mdx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+started with a controller. That is, a `controller` block and a `purpose = "api"`
+listener must be defined in Boundary's configuration file. Additionally, an
+`purpose = "ops"` listener must also be defined in Boundary's configuration
+file. Under these conditions the ops server (which hosts the controller health
```
Comment (taoism4504) [inline website/content/docs/operations/health.mdx:21]:
```suggestion
file. Under these conditions, the ops server (which hosts the controller health
```

File: website/content/docs/operations/health.mdx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+Boundary provides health monitoring through the `/health` path using a listener
+with the `"ops"` purpose. By default a listener with that purpose runs on port
+`9203`. See the [example configuration](#example-configuration) section for an
+example listener stanza in a `config.hcl` file.
+
+### Requirements
+
+For the controller health endpoint to be enabled, any Boundary instance must be
+started with a controller. That is, a `controller` block and a `purpose = "api"`
+listener must be defined in Boundary's configuration file. Additionally, an
```
Comment (taoism4504) [inline website/content/docs/operations/health.mdx:19]:
```suggestion
listener must be defined in Boundary's configuration file. Additionally, a
```

File: website/content/docs/operations/health.mdx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+`purpose = "ops"` listener must also be defined in Boundary's configuration
+file. Under these conditions the ops server (which hosts the controller health
+api) will be exposed.
+
+### Shutdown Grace Period
+
+Optionally, when the controller health endpoint is enabled Boundary can be
```
Comment (taoism4504) [inline website/content/docs/operations/health.mdx]: I'd like to suggest a change with this sentence to break it up into multiple sentences so that it reads well.  Please feel free to nix my suggestion if it doesn't make sense or I've somehow changed the intent of the sentence. First off, typically it's best to avoid starting a sentence off with 'Optionally.' 

Here's my suggestion:
When the controller health endpoint is enabled, Boundary may be configured to change the controller health response to a`403 Server Unavailable`. The controller health response will only change upon receiving a shutdown signal. An amount of time may be configured before the shutdown process begins.

File: website/content/docs/operations/health.mdx [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+### Requirements
+
+To enable the controller health endpoint, any Boundary instance must be
+started with a controller. That is, a `controller` block and a `purpose = "api"`
+listener must be defined in Boundary's configuration file. Additionally, a
+`purpose = "ops"` listener must also be defined in Boundary's configuration
+file. Under these conditions, the `ops` server (which hosts the controller health
+api) will be exposed.
+
+### Shutdown Grace Period
```
Comment (taoism4504) [inline website/content/docs/operations/health.mdx:24]: For the first sentence, my suggestion is to break it out into multiple sentences. Here's my suggestion, but please feel free to nix if my attempt to provide more clarity changed the intent of the sentence. Also, we'd want to avoid starting out a sentence with "Optionally."

**Suggestion**:
When the controller health endpoint is enabled, Boundary may be configured to change the controller health response to `503 Service Unavailable`. The controller health response will only change upon receiving a shutdown signal. An amount of time may be configured before the shutdown process begins.

File: website/content/docs/operations/health.mdx [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+upon receiving a shutdown signal, and wait a configurable amount of time before
+starting the shutdown process.
+ 
+This feature is designed to integrate with load balancers to reduce the risk of
+an outgoing Boundary instance causing disruption to incoming requests.
+ 
+In this state, Boundary is still capable of processing requests as normal, but
+will report as unhealthy through the controller health endpoint. In
+load-balanced environments, this would cause this "unhealthy" instance to be
+removed from the pool of instances eligible to handle requests, reducing the
```
Comment (taoism4504) [inline website/content/docs/operations/health.mdx:37]:
```suggestion
removed from the pool of instances eligible to handle requests, and thereby, reducing the
```

File: website/content/docs/operations/index.mdx [hunk 10]
```diff
@@ -0,0 +1,10 @@
+---
+layout: docs
+page_title: Operating Boundary
+description: |-
+  Operational tasks in Boundary.
+---
+
+# Maintaining and Operating Boundary
+
+Having visibility into various components of Boundary plays an important role in ensuring the health of its Controllers and Workers. While we intend to offer additional monitoring components in subsequent Boundary releases, Boundary can now export metrics for its Controllers and Workers which can be collected by Prometheus.  These metrics help you understand the current state of your Boundary deployment, thereby, enabling the diagnosis of deployed Controllers and Workers to help detect anomalies and mitigate. This section will cover the metrics available as well as the naming and labeling standards used for the metrics exported.
```
Comment (taoism4504) [inline website/content/docs/operations/index.mdx:10]:
```suggestion
Having visibility into various components of Boundary plays an important role in ensuring the health of its Controllers and Workers. While we intend to offer additional monitoring components in subsequent Boundary releases, Boundary can now export metrics for its Controllers and Workers which can be collected by Prometheus.  These metrics help you understand the current state of your Boundary deployment, thereby, enabling the diagnosis of deployed controllers and workers to help detect anomalies and mitigate. This section will cover the metrics available as well as the naming and labeling standards used for the metrics exported.
```

Comment (taoism4504) [reply website/content/docs/operations/index.mdx:10]: I've seen these terms used as lowercase so changing to lowercase for consistency.

Comment (taoism4504) [reply website/content/docs/operations/index.mdx:10]: Since we've added the health endpoint, I may modify the intro slightly to include a few sentences about that.

File: website/content/docs/operations/metrics.mdx [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+Boundary deployment through Prometheus.
+
+The metrics uses the [OpenMetric
+exposition](https://prometheus.io/docs/instrumenting/exposition_formats/#openmetrics-text-format)
+format which can be read by
+[Prometheus](https://prometheus.io/docs/introduction/overview/). 
+
+### Controller
+
+The following lists the current names and descriptions of metrics provided for controllers.
```
Comment (A-440Hz) [inline website/content/docs/operations/metrics.mdx:27]: nit: how about "The following table lists...."

File: website/content/docs/operations/metrics.mdx [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+| Name                                                          | Description                                   |
+|---------------------------------------------------------------|-----------------------------------------------|
+| `boundary_controller_api_http_request_duration_seconds`       | Historgram of latencies for HTTP requests.     |  
+| `boundary_controller_api_http_request_size_bytes`             | Histogram of request sizes for HTTP requests.  |
+| `boundary_controller_api_http_response_size_bytes`            | Histogram of response sizes for HTTP requests. |
+| `boundary_controller_cluster_grpc_request_duration_seconds`   | Histogram of latencies for requests made to the gRPC service running on the cluster listener. |
+
+### Worker
+
+The following lists the current names and descriptions of metrics provided for
```
Comment (A-440Hz) [inline website/content/docs/operations/metrics.mdx:38]: same as above

Comment (yhyakuna) [reply website/content/docs/operations/metrics.mdx]: Fixed ([commit](https://github.com/hashicorp/boundary/pull/2024/commits/1c24d17521cbcfdd6892118dc23e8019544b4a53))

File: website/content/docs/operations/metrics.mdx (hunk 11 above)
Comment (yhyakuna) [reply website/content/docs/operations/metrics.mdx]: Fixed ([commit](https://github.com/hashicorp/boundary/pull/2024/commits/1c24d17521cbcfdd6892118dc23e8019544b4a53))

File: website/content/docs/operations/index.mdx [hunk 13]
```diff
@@ -0,0 +1,13 @@
+---
+layout: docs
+page_title: Operating Boundary
+description: |-
+  Operational tasks in Boundary.
+---
+
+# Maintaining and Operating Boundary
+
+Having visibility into various components of Boundary plays an important role in ensuring the health of its controllers and workers. While we intend to offer additional monitoring components in subsequent Boundary releases, Boundary can now export metrics for its controllers and workers which can be collected by Prometheus.  These metrics help you understand the current state of your Boundary deployment, thereby, enabling the diagnosis of deployed controllers and workers to help detect anomalies and mitigate. We will cover the metrics available as well as the naming and labeling standards used for the metrics exported. Additionally, we offer the ability to monitor the health of your Boundary instance with new endpoints. Once you defined a listener in our Boundary's configuration file, the controller health endpoint becomes enabled, and a health response may be configured to begin a shutdown process that's based on a configurable amount of time.
```
Comment (johanbrandhorst) [inline website/content/docs/operations/index.mdx:10]: > Once you defined a listener in our Boundary's configuration file, the controller health endpoint becomes enabled, and a health response may be configured to begin a shutdown process that's based on a configurable amount of time.

It looks like we're missing a **have** in the first part of this sentence? Should `enabled` be `available`? The last part of the sentence implies to me that the health response can itself be configured to begin a shutdown process, but I don't think that's right.

File: website/content/docs/operations/metrics.mdx [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+---
+
+## Boundary Metrics
+
+Boundary provides metrics through the `/metrics` path using a listener with the
+`"ops"` purpose. By default, a listener with that purpose runs on port `9203`.
+See the [example configuration](#example-configuration) section for an example
+listener stanza in a `config.hcl` file.
+
+Metrics are important for understanding what is happening in a boundary
```
Comment (johanbrandhorst) [inline website/content/docs/operations/metrics.mdx:15]:
```suggestion
Metrics are important for understanding what is happening in a Boundary
```

File: website/content/docs/operations/metrics.mdx [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+[Prometheus](https://prometheus.io/docs/introduction/overview/). 
+
+### Controller
+
+The following table lists the current names and descriptions of metrics provided
+for controllers.
+
+| Name                                                          | Description                                   |
+|---------------------------------------------------------------|-----------------------------------------------|
+| `boundary_controller_api_http_request_duration_seconds`       | Historgram of latencies for HTTP requests.     |
```
Comment (johanbrandhorst) [inline website/content/docs/operations/metrics.mdx:32]:
```suggestion
| `boundary_controller_api_http_request_duration_seconds`       | Histogram of latencies for HTTP requests.     |  
```

File: website/content/docs/operations/metrics.mdx [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+
+The metrics uses the [OpenMetric
+exposition](https://prometheus.io/docs/instrumenting/exposition_formats/#openmetrics-text-format)
+format which can be read by
+[Prometheus](https://prometheus.io/docs/introduction/overview/). 
+
+### Controller
+
+The following table lists the current names and descriptions of metrics provided
+for controllers.
```
Comment (johanbrandhorst) [inline website/content/docs/operations/metrics.mdx:28]:
```suggestion
by controllers.
```

Should this be `by`?

File: website/content/docs/operations/metrics.mdx [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+|---------------------------------------------------------------|-----------------------------------------------|
+| `boundary_controller_api_http_request_duration_seconds`       | Historgram of latencies for HTTP requests.     |  
+| `boundary_controller_api_http_request_size_bytes`             | Histogram of request sizes for HTTP requests.  |
+| `boundary_controller_api_http_response_size_bytes`            | Histogram of response sizes for HTTP requests. |
+| `boundary_controller_cluster_grpc_request_duration_seconds`   | Histogram of latencies for requests made to the gRPC service running on the cluster listener. |
+
+### Worker
+
+The following table lists the current names and descriptions of metrics provided
+for workers.
```
Comment (johanbrandhorst) [inline website/content/docs/operations/metrics.mdx:40]:
```suggestion
by workers.
```
Should this be `by`?

File: website/content/docs/operations/metrics.mdx [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+
+| Name                                                          | Description                                   |
+|---------------------------------------------------------------|-----------------------------------------------|
+| `boundary_worker_proxy_http_write_header_duration_seconds`    | Histogram of time elapsed after the TLS connection is established to when the first http header is written back from the server. |
+| `boundary_worker_proxy_websocket_active_connections`          | A gauge of the current count of open proxy connections on the worker. |
+| `boundary_worker_proxy_websocket_received_bytes_total`        | Count of received bytes sent over all proxy connections handled by the worker. |
+| `boundary_worker_proxy_websocket_sent_bytes_total`            | Count of sent bytes  sent over all proxy connections handled by the worker. |
```
Comment (johanbrandhorst) [inline website/content/docs/operations/metrics.mdx:47]:
```suggestion
| `boundary_worker_proxy_websocket_sent_bytes_total`            | Count of sent bytes sent over all proxy connections handled by the worker. |
```

File: website/content/docs/operations/metrics.mdx [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+
+### Other
+
+The following table lists the current miscellaneous metrics provided for Boundary.
```
Comment (johanbrandhorst) [inline website/content/docs/operations/metrics.mdx:51]:
```suggestion
The following table lists the current miscellaneous metrics provided by Boundary.
```
Should this be `by`?

File: node_modules/.package-lock.json [hunk 20]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (tmessi) [inline node_modules/.package-lock.json:1]: Is this file needed?  This seems like it shouldn't be here

File: website/content/docs/release-notes/v0_8_0.mdx [hunk 21]
```diff
@@ -0,0 +1,33 @@
...
+
+Lastly, for instructions on how to upgrade an existing Boundary deployment to v0.8.0, please review Boundary’s [general upgrade guide](https://learn.hashicorp.com/tutorials/boundary/upgrade-version).
+
+## Boundary v0.8.0 Highlights
+
+**Metrics and Health Endpoint:** Boundary is adding [Prometheus metrics](https://www.boundaryproject.io/docs/operations/metrics) to monitor the operations of workers and controllers, 
+as well as a [health endpoint](https://www.boundaryproject.io/docs/operations/health) for controllers. 
+
+**Audit Event Logs:** We are expanding the audit event logs by increasing the captured [audit events](https://www.boundaryproject.io/docs/configuration/events/overview), 
+with support for sanitizing sensitive information from audit events. All of the events are now classified and events contain more usable data, with sensitive information encrypted by default.
```
Comment (johanbrandhorst) [inline website/content/docs/release-notes/v0_8_0.mdx:22]: Sensitive information is _redacted_ by default.

**Review by tmessi — DISMISSED** [review]

**Comment by johanbrandhorst** [comment]
https://github.com/hashicorp/boundary/pull/2024#discussion_r863177216 has not been responded to

**Comment by stellarsquall** [comment]
> [#2024 (comment)](https://github.com/hashicorp/boundary/pull/2024#discussion_r863177216) has not been responded to

I cleanup up this description quite a bit, would love your opinion

**Review by xingluw — APPROVED** [review]

**Review by GorbachovMaina1 — APPROVED** [review]
- [ ]
